File: 1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        x = {}
        for i,num in enumerate(nums):
            z = target - num
            if z not in x:
                x[num] = i
            else:
                return x[z],i

File: 1.Two sum.java
/* this can be done in multiple ways... if u want i can put some more approaches (write a issue)
Using HashMap, it takes >1ms i.e faster than 99.96% submissions on leetcode */
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //using hashmap
        Map<Integer,Integer> myMap = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int comp = target-nums[i];
            if(myMap.containsKey(comp)){
                return new int[] {myMap.get(comp),i};
            }
            myMap.put(nums[i],i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}
